Views/Widget-Link.liquid

```diff
@@ -1,20 +1,34 @@
-{% assign linkUrl = Model.ContentItem.Content.Link.Url.Text %}
+{% assign linkUrl = Model.ContentItem.Content.LinkDestinationPart.DestinationUrl.Text %}
+{% assign openIn = Model.ContentItem.Content.LinkBehaviourPart.OpenIn.Text %}
+{% assign isModal = openIn == 'modal' or openIn == 'modal-wide' %}
+
 {% assign text = Model.ContentItem.Content.Link.Text.Text %}
-{% assign openInNewWindow = Model.ContentItem.Content.Link.OpenInNewWindow.Value %}
 {% assign cssClasses = '' %}
-{% assign onClickEvent = Model.ContentItem.Content.JavaScriptEventsPart.OnClick.Value %}
+
+{% assign onClickEvent = Model.ContentItem.Content.LinkBehaviourPart.ClickEvent.Value %}
 
 {% comment %} "Style" field is obsolete but remains for backwards compatibility {% endcomment %}
-{% assign style = Model.ContentItem.Content.Link.Style.Text %}
+{% assign oldStyle = Model.ContentItem.Content.Link.Style.Text %}
 
 {% if Model.ContentItem.Content.LinkVisualPart.Style.Text %}
     {% assign style = Model.ContentItem.Content.LinkVisualPart.Style.Text %}
 {% endif %}
 
-{% if style != null and style != 'default' %}
+{% if oldStyle != null and oldStyle != 'default' %}
     {% assign cssClasses = 'btn ' | append: style %}

```

Comment:
Should this use `oldStyle`? Why check to see whether `oldStyle` is present and then append the new style value.

Reply:
This was already in the template, i just renamed the var because it was similar to the style var we have. 

i dont fully understand, what you mean

Reply:
You changed it to use `oldStyle`? Why? My assumption is that the `if` statement is checking whether `oldStyle` is present, and if so it's added to the `cssClasses`. However, the code currently is checking if `oldStyle` is present, and then adding `style` to the `cssClasses`.